docs/developers-guides/DevelopACustomImportPlugin.md

```diff
@@ -0,0 +1,85 @@
+# Develop a custom import plugin
+
+This documentation guides on creating a custom import plugin for ReportPortal, using the PF4J plugin framework.
+
+## Prerequisites
+Before starting the development of a new import plugin, ensure you review these essential resources:
+
+1. **PF4J Plugin Framework**: Learn about PF4J, which is the core framework for ReportPortal plugins system at [pf4j.org](https://pf4j.org).
+2. **Plugin Template**: Start with the standard plugin template available on [GitHub](https://github.com/reportportal/plugin-template).
+3. **Examples**: Examine existing plugins for importing:
+   - JUnit: [GitHub repo](https://github.com/reportportal/plugin-import-junit)
+   - RobotFramework: [GitHub repo](https://github.com/reportportal/plugin-import-robot).
+4. **Event-Based Reporting**: Understand event-based interactions in ReportPortal through [service-api](https://github.com/reportportal/service-api/blob/develop/src/main/java/com/epam/ta/reportportal/reporting/event/EventBasedReporting.java).
+5. **Event Objects**: Familiarize with event objects used in reporting at [GitHub](https://github.com/reportportal/commons/tree/develop/src/main/java/com/epam/reportportal/events).
+6. **Start Import Endpoint**: Learn how to initiate importing through the UI at [GitHub](https://github.com/reportportal/service-api/blob/6d40590d21fef3299ed5b7e009b1cc7130863c21/src/main/java/com/epam/ta/reportportal/ws/controller/PluginController.java#L140).
+
+## Step-by-Step Plugin Development Guide
+
+### Step 1: Clone the Plugin Template
+Begin by using the [plugin template](https://github.com/new?template_name=plugin-template&template_owner=reportportal) to create a new repository and get the required boilerplate code.
+
+### Step 2: Rename and Configure the Template
+Update the naming from 'template' to something specific to your plugin need. Modify the `gradle.properties` to set your `pluginId`, which should be a key word without special symbols. Update the manifest appropriately [here](https://github.com/reportportal/plugin-template/blob/698a17a862eb4a8d52274242a9f8e4ec9e6ab799/plugin/build.gradle#L101).
+
+### Step 3: Implement the Plugin EntryPoint
+Create your main class (or use from the template) annotated with `@Extension` that implements `ReportPortalExtensionPoint` ([TemplatePluginExtension](https://github.com/reportportal/plugin-template/blob/main/plugin/src/main/java/com/epam/reportportal/extension/template/TemplatePluginExtension.java)).
+This class should contain all logic for integrating the new plugin with the ReportPortal system. Implement the necessary methods as shown in the earlier example snippet.
+
+For correct representation on the UI, there should be added specific parameters to the plugin:
+
+```java
+  @Override
+  public Map<String, ?> getPluginParams() {
+    Map<String, Object> params = new HashMap<>();
+    params.put(ALLOWED_COMMANDS, new ArrayList<>(pluginCommandMapping.get().keySet()));
```

Comment:
As I see we don't use the `allowedCommands` in Junit and Robot import plugins, so I suppose it can be removed here.